share/arno-iptables-firewall/environment

```diff
@@ -320,18 +320,25 @@ ip4tables()
 {
   local result retval IFS=' '
 
-  result=`$IP4TABLES${IPTABLES_OPTIONS:+ $IPTABLES_OPTIONS} "$@" 2>&1`
+  exec 3>&1
+  err_result=`$IP4TABLES${IPTABLES_OPTIONS:+ $IPTABLES_OPTIONS} "$@" 2>&1 1>&3 |grep -v -e 'WARNING:.*match is obsolete' -e 'iptables-legacy tables present'`
   retval=$?
 
   if [ $retval -ne 0 ]; then
     # Show any (error) messages in red
-    printf "\033[40m\033[1;31m${IP4TABLES} $*\nERROR ($retval): ${result}\033[0m\n" >&2
```

Comment:
Wouldn't simply renaming `result` -> `err_result` be the same as all the new code ?

Reply:
No ;-) The idea is that we no longer redirect stderr to stdout and then process only stdout (which is too error prone, as it turns out). So now we just pass through stdout (via &3) and then redirect stderr to stdout and process then only stderr. So it's certainly different.

Reply:
No `:-)` (I wasn't clear) I was only referring to the one red line before this comment and the 10 green lines after this comment ... except for a added newline, the output looks the same to me.

Reply:
Well the idea is that when iptables returns 0 ("success"), its stderr still gets send to stderr instead of stdout (like we previously did). Have a close look at the usages of retval here.

Reply:
Ahh, I see it now (sorry for the delay `:-)`)

Possibly this is less change and more clear ... you decide:
```
  if [ $retval -ne 0 ]; then
    # Show any (error) messages in red
    printf "\033[40m\033[1;31m${IP4TABLES} $*\nERROR ($retval): ${err_result}\033[0m\n" >&2
    if note_iptables_error "$@"; then
      RULE_WARNING=$((RULE_WARNING + 1))
    fi
  elif [ -n "$err_result" ]; then
    # Success, but with stdout
    echo "$err_result" >&2
  fi
```

Reply:
Yes, I think you're right. That looks cleaner :-)